packages/runtime/src/core.ts

```diff
@@ -134,6 +150,13 @@ export class FederationHost {
     this.options = this.formatOptions(defaultOptions, userOptions);
   }
 
+  private envPlugin() {
+    if (typeof document === 'undefined') {
```

Comment:
you can use [inBrowser](https://github.com/module-federation/core/pull/2826/files#diff-05e0fa4ccdcd38297f77d08151fce61d702a14d79ea0940b2d8b4d1bbc3c2fdaL121) value to judge the env